Sources/SwiftParser/Names.swift

```diff
@@ -68,6 +213,9 @@ extension Parser {
   }
 
   mutating func parseDeclReferenceExpr(_ flags: DeclNameOptions = []) -> RawDeclReferenceExprSyntax {
+    // Consume a module selector if present.
+    let moduleSelector = self.parseModuleSelector()
```

Comment:
I think presence of the module selector should affect `flags` (`DeclNameOptions.keywords`).
It would be weird if `ModName::default` is rejected, but `foo.ModName::default` is allowed.
related: https://forums.swift.org/t/pitch-module-selectors/80835/82